### PR TITLE
[Serve] Add gauge to track number of control loops

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change adds a `Gauge` metric that tracks the number of control loops performed by the controller. It increases monotonically, and it gets reset to 0 when the controller restarts.

## Related issue number

N/A

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - This change relies on existing tests.

### CODE REVIEW

File: python/ray/serve/controller.py
Comment (edoakes) [inline python/ray/serve/controller.py:416]: can you add the controller's actor_id as a tag on the gauge?

Comment (shrekris-anyscale) [reply python/ray/serve/controller.py:416]: I added the `actor_id`.

**Comment by shrekris-anyscale** [comment]
The test failures are unrelated:

* `semgrep` is failing on `master`:

Screen Shot 2023-08-03 at 4 21 07 PM

**Comment by shrekris-anyscale** [comment]
@edoakes This change is ready to merge pending approval.

**Review by edoakes — APPROVED** [review]

**Pull request merged**
